### PR TITLE
feat(split-button): describe split button using design tokens (FE-4448)

### PR DESCRIPTION
**Pull request by mjaniczko-sage**
### Proposed behaviour
Describes the SplitButton using design tokens. Removes all possible 'theme' in css files

Changes made to match new styles (https://www.figma.com/file/Fp6Vn7X28ShX99IYDKH9rc/ARCHIVE-Sage-DS---Official-Library?node-id=60%3A17777):
Background color for open 
Backgournd:hover color for open
Color for open 

### Current behaviour
The SplitButton use the theme styled-component property. 
Use old styles

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required

#### QA

- [ ] Tested in CodeSandbox/storybook
- [ ] Add new Cypress test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.

### CODE REVIEW

**Review by jhajduk-sage — DISMISSED** [review]

**Review by grabkowski — APPROVED** [review]
Looks fine, could you squash all commits into one?

**Review by grabkowski — DISMISSED** [review]

**Review by DipperTheDan — DISMISSED** [review]

**Comment by harpalsingh** [comment]
Screenshot 2022-01-05 at 09 12 57

There should not be a white border on the primary split button, but the Figma file doesnt seem to work, i will confirm the value.

**Comment by harpalsingh** [comment]
Had a look at the design provided and while we may review that, the design shows a 2px gap above and below the white seperator, this is not implemented correctly.

Screenshot 2022-01-05 at 09 22 03

**Review by grabkowski — APPROVED** [review]

File: src/components/split-button/split-button-toggle.style.js [hunk 1]
```diff
@@ -49,3 +59,5 @@ const StyledSplitButtonToggle = styled(StyledButton)`
 `;
 
 export default StyledSplitButtonToggle;
+
+// `border-left-color: var(--colorsActionMajorYang100);`
```
Comment (DipperTheDan) [inline src/components/split-button/split-button-toggle.style.js:63]: Why is this code commented out? Should this have been removed?

**Review by DipperTheDan — DISMISSED** [review]

**Review by grabkowski — APPROVED** [review]

**Review by Parsium — APPROVED** [review]

**Comment by harpalsingh** [comment]
Could we add the same shadow multiaction button uses on the open menu pls.

Screenshot 2022-01-18 at 09 24 46

**Review by Parsium — APPROVED** [review]

**Review by DipperTheDan — APPROVED** [review]

**Review by ZhuoyuJin — APPROVED** [review]

**Pull request merged**